modules/gapi/samples/pipeline_modeling_tool/utils.hpp

```diff
@@ -14,6 +14,20 @@ struct OutputDescr {
 };
 
 namespace utils {
+
+inline void createNDMat(cv::Mat& mat, const std::vector<int>& dims, int depth) {
+    GAPI_Assert(!dims.empty());
+    // FIXME: Now G-API doesn't create type for ND cv::Mat's, so it's needed to
+    // create cv::Mat the same way in order to achieve meta matching.
+    // Change this line when it's fixed to:
+    // mat.create(dims, CV_MAKETYPE(depth, dims.size()));
+    mat.create(dims, depth);
```

Comment:
@mshabunin @alalek Could you help us with understanding?
Which line is correct to create ND `cv::Mat`?

Reply:
> mat.create(dims, CV_MAKETYPE(depth, dims.size()));

This doesn't look valid as `dims.size()` should NOT be used as number of channels.

---

> `mat.create(dims, depth);`

This code is equal to `mat.create(dims, CV_MAKETYPE(depth, 1));`

Reply:
And what's the correct way to create ND `Mat` with dims and let's say `CV_32F` depth?

```
mat.create(dims, CV_32f)
```

Reply:
`CV_32F` is allowed (CN=1 is assumed).
`CV_32FC1` is accurate.